### ISSUE TITLE
Feature request: Add support for biome as an alternative to eslint/prettier

### ISSUE THREAD

**Issue by WtfJoke** [projen/projen#3966]
We switched all our projects from eslint and prettier to biome. 

https://biomejs.dev/ is a tool which can be used as a replacement for either prettier or eslint/typescript-eslint or both.
So biome can lint or format or both. Its much faster and the configuration is much easier (at least in my opinion).

It would be great if we can configure projen to use/configure biome.

I could imagine a config like (similar to github/githuboptions):
```jsonc
biome: true // to activate both formatting + linting
biomeOptions: { // optional
  format: false // false=would still use prettier for formatting
  lint: false // false=would still use eslint for linting
  // .. additional options from https://biomejs.dev/reference/configuration/
}
```

<details>
  <summary>Our current `biome.jsonc` looks like this (just for reference):</summary>

```jsonc
{
    "$schema": "https://biomejs.dev/schemas/1.8.3/schema.json",
    "vcs": {
        "enabled": true,
        "clientKind": "git",
        "useIgnoreFile": true
    },
    "files": {
        "ignore": [
            // projen managed files
            ".projen/",
            ".projenrc.ts",
            "tsconfig*.json",
            "package.json",
            // end of projen managed files
            "*test.ts.snapshot/**" // cdk integTest snapshots
        ]
    },
    "formatter": {
        "enabled": true,
        "formatWithErrors": false,
        "indentStyle": "space",
        "indentWidth": 4,
        "lineEnding": "lf",
        "lineWidth": 120,
        "attributePosition": "auto"
    },
    "organizeImports": { "enabled": true },
    "linter": {
        "enabled": true,
        "rules": {
            "recommended": true,
            "correctness": {
                "noUnusedImports": "warn"
            }
        }
    },
    "overrides": [
        {
            "include": ["**/*.test.ts"],
            "linter": {
                "rules": {
                    "style": {
                        "noNonNullAssertion": "off"
                    },
                    "suspicious": {
                        "noExplicitAny": "off"
                    }
                }
            }
        }
    ]
}
```
</details>

**Comment by mrgrain** [comment]
Great suggestion. Just to manage expectations, you'll probably have to contribute this yourself.

**Comment by WtfJoke** [comment]
> Great suggestion. Just to manage expectations, you'll probably have to contribute this yourself.

Yeah thats fine. :) I am not planning to do that in near future. 
However I wanted to at least contribute by reporting all the issues I've encountered.

**Comment by Malanius** [comment]
Started switching to biome as well. Support for this would be useful. Could try to hack it in myself if I have some free time. Any pointers on how to do or approach this?

**Comment by Hi-Fi** [comment]
Just created component for Biome: https://github.com/Hi-Fi/projen-biome.

Not sure which one should be the way with Projen; try to get all to this repo, or offer just bare minimum and then utilize components from other repos. 

Of course Projen's core can wrap those external components if it would be easier than adding classes to projenrc.

**Comment by mrgrain** [comment]
Honestly looks pretty solid already @Hi-Fi and we would happily accept a PR. I'm not entirely sure I'm following all the stuff for biome + eslint and biome + prettier (I thought biome replaced these), but that can probably be iterated. 

I think the only "improvement" I can see would be to the schema import. Probably prefer an explicit GH Action task that copies the schema file under projen's version control.

**Comment by Hi-Fi** [comment]
Biome can replace both eslint and prettier (and maybe even should). Currently those are separated so configs were taken to both.

Schema part could be done with explicit action, and it shouldn't be even need to be run as often as it currently is in the repo. As long as major doesn't change old types would be OK unless something specific comes. So not sure if action would be the way or some dynamic fetching if something else than known version if used.

So @mrgrain preferred way would be that projen repo would contain those components?

**Comment by mrgrain** [comment]
> Biome can replace both eslint and prettier (and maybe even should).

Gotcha, so the 'can' is essential here. The current eslint/prettier setup is complementary. Really projen should have a `formatter` and a `linter` option. We will need to think about how to add biome now.

> So @mrgrain preferred way would be that projen repo would contain those components?

Yes, I don't see a reason why not. There is a long term future that would have a `@projen/biome` package - but the current way of handling things doesn't support that yet.

**Comment by Hi-Fi** [comment]
I tried to make configuration same way that user can select either-or (either formatting or linting). It's just that clear always as linter also does formatting...

But I'll check if I'm able to make PR based on the linked repo.

**Comment by Hi-Fi** [comment]
@dannysteenman true, biome seems to be stricter on directories than eslint, or then those first were for projen itself.

PR is fastest way, not sure if project itself would prefer another issue, too.

**Comment by dannysteenman** [comment]
> [@dannysteenman](https://github.com/dannysteenman) true, biome seems to be stricter on directories than eslint, or then those first were for projen itself.
> 
> PR is fastest way, not sure if project itself would prefer another issue, too.
> 
> Root with ignores might also work. Nicest would be dynamic enough to work with various project types (like CDK projects usually use bin and lib instead of src)

sure! @mrgrain what do you think and what would you prefer?
